airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -386,10 +386,14 @@ def execute(self, context: 'Context'):
                 result = self.extract_xcom(pod=self.pod)
             remote_pod = self.pod_manager.await_pod_completion(self.pod)
         finally:
-            self.cleanup(
-                pod=self.pod or self.pod_request_obj,
-                remote_pod=remote_pod,
-            )
+            pod = self.pod or self.pod_request_obj
+            if self.find_pod(pod.metadata.namespace, context=context):
+                self.cleanup(
+                    pod=pod,
+                    remote_pod=remote_pod,
+                )
+            else:
+                self.log.error("Could not find pod with name: %s and namespace: %s", pod.metadata.name, pod.metadata.namespace)
```

Comment:
We should probably say what we didn't do since we didn't find the pod, and it can't just be based on name, it's for this TI. Maybe something along these lines (we could probably be more verbose if we want)?

`Could not find pod for this TI to delete`